libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -226,39 +264,76 @@ void AC_PrecLand::handle_msg(mavlink_message_t* msg)
     }
 }
 
+// send_landing_target - Send LANDING_TARGET mavlink message, called from ArduCopter/GCS_Mavlink.cpp as part of EXTRA3 stream
+void AC_PrecLand::send_landing_target(mavlink_channel_t chan)
+{
+    // If we have the sensor timestamp then set that as time_usec, otherwise return measurement timestamp
+    uint64_t time_usec;
+    if (_last_backend_los_timestamp) {
+        time_usec = _last_backend_los_timestamp;
+    } else {
+        time_usec = _last_backend_los_meas_ms * 1000;
+    }
+
+    // Pack and send the LANDING_TARGET message
+    // Note: TODO are for message fields that do not yet have data members in AC_PrecLand class or are not supported yet
+    // Note: 0 values here are sent in absence of sensible defaults.  They are not correct, but there is no obvious alternative
+    mavlink_msg_landing_target_send(
+        chan,
+        time_usec, // sensor/measurement timestamp in microseconds, either from epoch or since boot
+        0, // TODO: Target ID
+        MAV_FRAME_BODY_NED, // frame
+        _target_pos_rel_est_NE.x * 100.0f, // angle_x,
```

Comment:
_target_pos_rel_est_NE is more x,y,z pose than angle. 
I think we should put _target_pos_rel_meas_NED in x,y,z

Reply:
If you look in both the Raw and EKF estimators, the last thing they do before handing off to the output prediction is to copy the _target_pos_rel_meas_NED (in the case of Raw) or _ekf_x and _ekf_y (in the case of EKF) to _target_pos_rel_est_NE, which is used to calculate the output prediction.  So it's the final step of data, I thought it was probably the best to add to the message?

Reply:
Actually wonder if we should use _target_pos_rel_out_NE instead, as that's been through the forward prediction and the offset corrections?

Reply:
@khancyr Yes I think you're right, have changed back to _target_pos_rel_meas_NED - we can use x,y,z and q later for processed data.

Reply:
hummm I think you were right, we should use _target_pos_rel_out_NE as it is the one with the correct orientation in the vehicle frame but I maintain tha is should be in x,y,z not angle. According to mavlink specs , angle is for angular offset of the target from the center of the image

Reply:
@khancyr what about this?
https://github.com/ArduPilot/ardupilot/pull/9125/commits/525dda3b263d1a3eae6ca73da8b21a1f402fb820
My trig sucks so the maths needs to be verified, but the idea is to calculate the angular offsets using inverse functions from the backend sensor measurement and use that for landing_target.angle_x and landing_target.angle_y.

Reply:
Then we can use post-prediction positions (_target_pos_rel_out_NE) for xy, but it's only a Vector2f and doesn't contain z.
https://github.com/ArduPilot/ardupilot/pull/9125/commits/d557ec2b532cbad8e06482270e155ca4a86bbc7b